awx/main/credential_plugins/tss.py

```diff
@@ -1,7 +1,16 @@
 from .plugin import CredentialPlugin
 from django.utils.translation import gettext_lazy as _
 
-from thycotic.secrets.server import DomainPasswordGrantAuthorizer, PasswordGrantAuthorizer, SecretServer, ServerSecret
+try:
+    from delinea.secrets.server import DomainPasswordGrantAuthorizer, PasswordGrantAuthorizer, SecretServer, ServerSecret
+except ImportError:
+    try:
+        from thycotic.secrets.server import DomainPasswordGrantAuthorizer, PasswordGrantAuthorizer, SecretServer, ServerSecret
+    except ImportError:
+        SecretServer = None
+        ServerSecret = None
+        PasswordGrantAuthorizer = None
+        DomainPasswordGrantAuthorizer = None
```

Comment:
I know this is a common pattern in Ansible-core, but it just makes things harder to debug if the imports do go wrong, because instead of troubleshooting an import error, people are troubleshooting why a class is `None` for some reason.

However, I see some wisdom in the fallback to `thycotic` to be more flexible for someone who needed to make changes to their site-packages.

So I would prefer to get rid of these `= None` lines, and also add a brief comment explaining the libary's name change.

Reply:
@AlanCoding 
We can set flag **HAS_TSS_SDK = False** if tss SDK is not found while importing instead of assigning None. And if flag value is False then we will raise AnsibleError at the start of the function as below.

if not HAS_TSS_SDK:
        raise AnsibleError("python-tss-sdk must be installed to use this plugin")

what do you think about this?

CC: @tylerezimmerman

Reply:
I know that's the common Ansible-core pattern, but if you look at the other plugins in this folder none of them do that, I don't see what value it offers, it's hard to test, and it is easy to get it wrong.

Reply:
@AlanCoding 
You want only **"from delinea.secrets.server import DomainPasswordGrantAuthorizer, PasswordGrantAuthorizer, SecretServer, ServerSecret"** there instead of whole block of statement. right?

Reply:
I'm fine to keep the imports as:

```python
try:
    from delinea.secrets.server import DomainPasswordGrantAuthorizer, PasswordGrantAuthorizer, SecretServer, ServerSecret
except ImportError:
    from thycotic.secrets.server import DomainPasswordGrantAuthorizer, PasswordGrantAuthorizer, SecretServer, ServerSecret
```

I think there are valid reasons for that. I just don't want to get into this rabbit hole of potentially handling `None` values from the 2nd except block.

Reply:
@AlanCoding 
I have modified the code.